### ISSUE TITLE
config_*() methods overwrite previous attributes

### ISSUE THREAD

**Issue by jakevdp** [ellisonbg/altair#130]
Here is the current behavior:

```python
from altair import Chart, load_dataset
cars = load_dataset('cars')

chart = Chart(cars)
chart.configure_axis(axisWidth=500)
chart.configure_axis(axisColor='red')
chart.to_dict()['config']
# {'axis': {'axisColor': 'red'}}
```
Note that ``axisWidth`` is overwritten by the second method call. I'm not sure whether it makes more sense to have it this way, or to have it so that additional method calls add properties to those which were previously defined.

**Comment by ellisonbg** [comment]
I think it is pretty likely that folks will call the configure methods
multiple times and I think the current behavior is a bit surprising. So
yeah, maybe the merge/update approach is better.

On Mon, Jun 13, 2016 at 2:50 PM, Jake Vanderplas <notifications@github.com>
wrote:

> Here is the current behavior:
>
> from altair import Chart, load_dataset
> cars = load_dataset('cars')
>
> chart = Chart(cars)
> chart.configure_axis(axisWidth=500)
> chart.configure_axis(axisColor='red')
> chart.to_dict()['config']
>
> {'axis': {'axisColor': 'red'}}
>
> I'm not sure whether it makes sense to have it this way, or to have it so
> that additional method calls add properties to those which were previously
> defined.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ellisonbg/altair/issues/130>, or mute the thread
> <https://github.com/notifications/unsubscribe/AABr0FvuDq1iw_1fCVqW4LfyYbGBJXhaks5qLdC2gaJpZM4I0xXE>
> .
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by jakevdp** [comment]
OK. I have some prototype code that will handle this – I'll try to do a PR tonight.

**Comment by jakevdp** [comment]
Fixed in #133
